### ISSUE TITLE
Build Error - TypeError: Cannot read property 'hostedUIProviderCreds' of undefined

### ISSUE THREAD

**Issue by Caoimhin89** [aws-amplify/amplify-cli#5989]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
During the backend build stage, I get an error stating the backend could not be pulled successfully due to the error "Cannot read property 'hostedUIProviderCreds' of undefined. Here is the Build Log Output:

```
                                 # Starting phase: build
                                 # Executing command: amplifyPush --simple
2020-11-27T14:56:08.376Z [INFO]: 
2020-11-27T14:56:08.377Z [INFO]: For more information on AWS Profiles, see:
                                 https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
2020-11-27T14:56:08.394Z [WARNING]: (node:1962) ExperimentalWarning: The fs.promises API is experimental
2020-11-27T14:56:08.696Z [INFO]: Amplify AppID found: *********. Amplify App name is: my-app
2020-11-27T14:56:08.793Z [INFO]: Backend environment prod found in Amplify Console app: my-app
2020-11-27T14:56:09.342Z [INFO]: 
2020-11-27T14:56:09.345Z [WARNING]: - Fetching updates to backend environment: prod from the cloud.
2020-11-27T14:56:09.883Z [WARNING]: ✔ Successfully pulled backend environment prod from the cloud.
2020-11-27T14:56:09.942Z [WARNING]: ✖ There was an error initializing your environment.
2020-11-27T14:56:09.950Z [INFO]: Failed to pull the backend.
2020-11-27T14:56:09.952Z [INFO]: Cannot read property 'hostedUIProviderCreds' of undefined
2020-11-27T14:56:09.956Z [INFO]: TypeError: Cannot read property 'hostedUIProviderCreds' of undefined
                                 at recursiveOmit (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:30:33)
                                 at recursiveOmit (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:34:3)
                                 at recursiveOmit (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:34:3)
                                 at recursiveOmit (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:34:3)
                                 at Object.exports.removeFromDeploymentSecrets (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:20:5)
                                 at AmplifyToolkit.saveEnvResourceParameters [as _saveEnvResourceParameters] (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/envResourceParams.ts:93:9)
                                 at /root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/index.js:301:23
                                 at process._tickCallback (internal/process/next_tick.js:68:7)
2020-11-27T14:56:09.972Z [ERROR]: !!! Build failed
2020-11-27T14:56:09.973Z [ERROR]: !!! Non-Zero Exit Code detected
2020-11-27T14:56:09.973Z [INFO]: # Starting environment caching...
2020-11-27T14:56:09.973Z [INFO]: # Environment caching completed
Terminating logging...
```

However, when I review the team-provider-info.json file I see that the hostedUIProviderCreds are present.
```
"prod": {
    "awscloudformation": {
      "AuthRoleName": "amplify-my-app-######-authRole",
      "UnauthRoleArn": "arn:aws:iam::############:role/amplify-my-app-######-unauthRole",
      "AuthRoleArn": "arn:aws:iam::############:role/amplify-my-app-######-authRole",
      "Region": "us-west-2",
      "DeploymentBucketName": "my-app-prod-######-deployment",
      "UnauthRoleName": "amplify-my-app-######-unauthRole",
      "StackName": "my-app-prod-######",
      "StackId": "arn:aws:cloudformation:us-****-*:######:stack/amplify-my-app-######/********-****-****-****-************",
      "AmplifyAppId": "my-app-id"
    },
    "categories": {
      "auth": {
        "myAppAuth": {
          "hostedUIProviderCreds": "[{\"ProviderName\":\"Facebook\",\"client_id\":\"**************\",\"client_secret\":\"********\"},{\"ProviderName\":\"Google\",\"client_id\":\"**********-*********.apps.googleusercontent.com\",\"client_secret\":\"*****-*****\"},{\"ProviderName\":\"LoginWithAmazon\",\"client_id\":\"amzn1.application-****-client.****************\",\"client_secret\":\"*******************************\"}]"
        },
        "userPoolGroups": {}
      },
      "function": {
        "lambda": {},
        "myCustomFuncFunc": {},
        "myOtherCustomFunc": {}
      }
    }
  }, 
```

**Amplify CLI Version**
v4.36.0

**To Reproduce**
amplifyPush --simple

**Expected behavior**
I expected the project to successfully pull the backend and complete the build successfully.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Amazon Linux (Amplify Console)
 - Node Version: 10.16.0

**Additional context**

**Comment by ammarkarachi** [comment]
@Caoimhin89 We have identified the bug we will have a fix soon.

**Comment by Caoimhin89** [comment]
@ammarkarachi That's great news! Thank you!

**Comment by ammarkarachi** [comment]
This should be fixed in the release 4.36.1

**Comment by Caoimhin89** [comment]
@ammarkarachi awesome, that fixed it! Thank you!
